AsyncDisplayKit/ASMapNode.mm

```diff
@@ -307,6 +318,15 @@ - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
   return size;
 }
 
+- (void)calculatedLayoutDidChange
+{
+  [super calculatedLayoutDidChange];
+  
+  if (_snapshotNeeded) {
+    [self takeSnapshot];
```

Comment:
@aaronschubert0 also curious -- I am moderately familiar with this code, but want to make sure I'm not overlooking a problem here.

It does seem to me like this could only increase the number of cases where a snapshot is taken.  Unless your comment on this line is actually referring to the lines above that gate inside the -takeSnapshot method, in which case let's focus on that thread.